networkit/cpp/graph/Graph.h

```diff
@@ -80,6 +83,7 @@ class Graph final {
 
 	friend class ParallelPartitionCoarsening;
 	friend class GraphBuilder;
+	friend class CurveballDetails::CurveballMaterialization;
```

Comment:
While this is OK for now, it would be desirable to extract the materialization class from the Curveball namespace - it is not really Curveball specific, right?

Reply:
Yes, it's very similar to the GraphBuild with the major difference that the GraphBuilder implements an in situ push interface, while this class takes an edge list. Also, the new implementation is much faster for our application. So following your suggestion I won't change it until the next release.